Tests/SwiftSyntaxTest/SyntaxConvenienceMethodsTests.swift

```diff
@@ -0,0 +1,22 @@
+import XCTest
+import SwiftSyntax
+
+public class SyntaxConvenienceMethodsTestCase: XCTestCase {
+
+  public static let allTests = [
+    ("testFloatingValue", testFloatingValue),
+    ("testIntegerValue", testIntegerValue),
+  ]
+
+  public func testFloatingValue() {
+    let digits = SyntaxFactory.makeFloatingLiteral("5.38")
```

Comment:
Ideally these would throw an error if the string value that is provided is not a floating/integer value, so you don't get a crash at a later point when trying to get the value from the node, beyond where the issue originated. But it doesn't have to be part of this PR.

Reply:
I thought it would feel odd if getting the integer value of an integer expression might fail. If something crashes, we must have done something seriously wrong in the SwiftSyntax parser, I imagine.

Reply:
I'm talking about `SyntaxFactory.makeFloatingLiteral` and `makeIntegerLiteral`. Currently you can pass any arbitrary string as parameters.

Reply:
Ah, OK. Will maybe tackle that later.